python/cufinufft/cufinufft.py

```diff
@@ -60,12 +60,12 @@ def __init__(self, nufft_type, modes, isign, tol,
         ready for point setting, and execution.
         """
 
-        # Note when None, opts will be populated with defaults by
-        # the library internally.  Advanced users may use
-        # `cufinufft.default_opts` to generate defaults and overload them
-        # before instantiating their cufinufft instances,
-        #  but this is currently undocumented.
-        self.opts = opts
+        if isign is None:
+            if nufft_type == 2:
+                isign = -1
+            else:
+                isign = +1
+
 
         # Setup type bound methods
         self.dtype = np.dtype(dtype)
```

Comment:
in FINUFFT
https://github.com/flatironinstitute/finufft/blob/b1a1f860fa974ae3295cd59c8cfe5bd2286e5f73/python/finufft/_interfaces.py#L97

you use more complicated way to check dtype in kwargs... but I guess it's the same thing.

Reply:
I don't think what we're doing in FINUFFT to get the dtype is necessary if we provide it as a regular argument. It will be removed from the `kwargs` dict and processed just like the other arguments. I'm guessing that code is a holdover from some earlier attempt at parsing the opts.